k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapFolderTest.java

```diff
@@ -507,6 +538,36 @@ public void getHighestUid() throws Exception {
     }
 
     @Test
+    public void getHighestUid_imapConnectionThrowsNegativesResponse_shouldReturnMinus1() throws Exception {
+        ImapFolder folder = createFolder("Folder");
+        prepareImapFolderForOpen(OPEN_MODE_RW);
+        List<ImapResponse> imapResponses = singletonList(createImapResponse("* SEARCH 42"));
+        when(imapConnection.executeSimpleCommand("UID SEARCH *:*"))
+                .thenThrow(NegativeImapResponseException.class);
+        folder.open(OPEN_MODE_RW);
+
+        long result = folder.getHighestUid();
+
+        assertEquals(-1L, result);
```

Comment:
Ah, I see, it's a convention. If you have a small test with only one assertion this is usually not a problem. 

I personally find the short form easier to read because I don't need to change focus / mentally switch between the lines when I read the code.

To make it a bit more readable with the arrange-act-assert pattern I'd suggest naming `result` to something more telling.

```java
long highestUid =  folder.getHighestUid();
assertEquals(-1L, highestUid);
```

The assert line is a little bit more self-documenting this way, `result` is quite generic.